tc/sgx/trusted_worker_manager/enclave/signup_enclave_kme.cpp

```diff
@@ -0,0 +1,317 @@
+
+        Zero(outPublicEnclaveData, inAllocatedPublicEnclaveDataSize);
+        Zero(outSealedEnclaveData, inAllocatedSealedEnclaveDataSize);
+
+        // Get instance of enclave data
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        enclaveData->set_extended_data((const char*) inExtData);
+
+        tcf::error::ThrowIf<tcf::error::ValueError>(
+            inAllocatedPublicEnclaveDataSize < enclaveData->get_public_data_size(),
+            "Public enclave data buffer size is too small");
+
+        tcf::error::ThrowIf<tcf::error::ValueError>(
+            inAllocatedSealedEnclaveDataSize < enclaveData->get_sealed_data_size(),
+            "Sealed enclave data buffer size is too small");
+
+        // Create the report data we want embedded in the enclave report.
+        sgx_report_data_t reportData = {0};
+        CreateSignupReportDataKME(inExtData, enclaveData, &reportData);
+
+        sgx_status_t ret = sgx_create_report(
+            inTargetInfo, &reportData, outEnclaveReport);
+        tcf::error::ThrowSgxError(ret, "Failed to create enclave report");
+
+        // Seal up the signup data into the caller's buffer.
+        // NOTE - the attributes mask 0xfffffffffffffff3 seems rather
+        // arbitrary, but according to Intel SGX SDK documentation, this is
+        // what sgx_seal_data uses, so it is good enough for us.
+        sgx_attributes_t attribute_mask = {0xfffffffffffffff3, 0};
+        ret = sgx_seal_data_ex(SGX_KEYPOLICY_MRENCLAVE, attribute_mask,
+            0,        // misc_mask
+            0,        // additional mac text length
+            nullptr,  // additional mac text
+            enclaveData->get_private_data_size(),
+            reinterpret_cast<const uint8_t*>(enclaveData->get_private_data().c_str()),
+            static_cast<uint32_t>(enclaveData->get_sealed_data_size()),
+            reinterpret_cast<sgx_sealed_data_t*>(outSealedEnclaveData));
+        tcf::error::ThrowSgxError(ret, "Failed to seal signup data");
+
+        // Give the caller a copy of the signing and encryption keys
+        // inAllocatedPublicEnclaveDataSize + 1 is to consider NULL character
+        strncpy_s(outPublicEnclaveData, inAllocatedPublicEnclaveDataSize+1,
+            enclaveData->get_public_data().c_str(),
+            enclaveData->get_public_data_size());
+    } catch (tcf::error::Error& e) {
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Error in Avalon enclave(ecall_CreateSignupDataKME): %04X -- %s",
+            e.error_code(), e.what());
+        ocall_SetErrorMessage(e.what());
+        result = e.error_code();
+    } catch (...) {
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Unknown error in Avalon enclave(ecall_CreateSignupDataKME)");
+        result = TCF_ERR_UNKNOWN;
+    }
+
+    return result;
+}  // ecall_CreateSignupDataKME
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_VerifyEnclaveInfoKME(const char* enclave_info, const char* mr_enclave,
```

Comment:
correct indentation for 2nd argument. Is cppcheck lint complaining about this ?

Reply:
Corrected indentations. cpplint didn't complain about this.